tests/e2e/koperator_test.go

```diff
@@ -0,0 +1,146 @@
+// existence afterwards.
+func requireApplyingKoperatorCRDs(kubectlOptions *k8s.KubectlOptions, koperatorVersion Version) {
+	It("Applying koperator CRDs", func() {
+		By(fmt.Sprintf("Retrieving Koperator CRDs (to work around too long CRD) with version %s ", koperatorVersion))
+		// Note: had to go lower because none of apply, create, replace works
+		// properly with the KafkaCluster CRD due to the size (apply cannot
+		// handle the size, create is not idempotent, replace is only idempotent
+		// if the resources are existing already). Tried dropping the
+		// descriptions in the CRDs, but still too large
+		var rawKoperatorCRDsManifest []byte
+		var err error
+
+		switch koperatorVersion {
+		case LocalVersion:
+			rawKoperatorCRDsManifest = []byte(helm.RenderTemplate(
+				GinkgoT(),
+				&helm.Options{
+					SetValues: map[string]string{
+						"crd.enabled": "true",
+					},
+				},
+				"../../charts/kafka-operator",
+				"dummy",
+				[]string{"templates/crds.yaml"},
+			))
+		default:
+			httpClient := new(http.Client)
+			httpClient.Timeout = 5 * time.Second
+
+			response, err := httpClient.Get("https://github.com/banzaicloud/koperator/releases/download/" + koperatorVersion + "/kafka-operator.crds.yaml")
+			if response != nil {
+				defer func() { _ = response.Body.Close() }()
+			}
+
+			Expect(err).NotTo(HaveOccurred())
+
+			rawKoperatorCRDsManifest, err = io.ReadAll(response.Body)
+
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		rawKoperatorCRDs := bytes.Split(rawKoperatorCRDsManifest, []byte("\n---\n"))
+		for rawKoperatorCRDIndex, rawKoperatorCRD := range rawKoperatorCRDs {
+			rawKoperatorCRDs[rawKoperatorCRDIndex] = bytes.Trim(rawKoperatorCRD, "\n-")
+		}
+
+		allowedCRDByteCount := 262144
+		tempDir := os.TempDir()
+		tempFile := path.Join(tempDir, "manifest.yaml")
+		for _, rawKoperatorCRD := range rawKoperatorCRDs {
+			err = os.WriteFile(tempFile, rawKoperatorCRD, 0o777)
+
+			Expect(err).NotTo(HaveOccurred())
+
+			if len(rawKoperatorCRD) > allowedCRDByteCount { // Note: too long CRDs cannot be applied, only created or replaced.
+				var koperatorCRD apiextensionsv1.CustomResourceDefinition
+				err := yaml.Unmarshal(rawKoperatorCRD, &koperatorCRD)
+
+				Expect(err).NotTo(HaveOccurred())
+
+				createOrReplaceK8sResourcesFromManifest(kubectlOptions, "crd", koperatorCRD.GetName(), tempFile, false)
+			} else {
+				By("Applying koperator CRDs with version " + koperatorVersion)
+				applyK8sResourceManifest(kubectlOptions, tempFile)
+			}
+		}
+
+		By("Verifying koperator CRDs")
+		requireExistingCRDs(
+			kubectlOptions,
+			"cruisecontroloperations.kafka.banzaicloud.io",
+			"kafkaclusters.kafka.banzaicloud.io",
+			"kafkatopics.kafka.banzaicloud.io",
+			"kafkausers.kafka.banzaicloud.io",
+		)
+	})
+}
+
+// requireInstallingKoperator deploys koperator CRDs and Helm chart and checks
+// the success of those operations.
+func requireInstallingKoperator(kubectlOptions *k8s.KubectlOptions, koperatorVersion Version) {
```

Comment:
I suggest to not use pointer type for kubectlOptions. 
I found that in some cases I needed to change the kubectlOptions.namespace value to "" to avoid an error or warning. I also had to change back the original value because of the pointer type.

Reply:
Maybe it would be be safer to use not pointer type for this. What do you think?

Reply:
Fixed in ef59837.

This does make it a bit more cumbersome to use kubectlOptions, but I agree the trade-off may be desirable.